chrome/content/zotero/preferences/preferences_sync.js

```diff
@@ -177,6 +177,129 @@ Zotero_Preferences.Sync = {
+		var indexOfId = syncedLibraries.indexOf(id);
+		if (indexOfId != -1) {
+			syncedLibraries.splice(indexOfId, 1);
+		} else {
+			syncedLibraries.push(id);
+		}
+		Zotero.Prefs.set('sync.syncedLibraries', JSON.stringify(syncedLibraries));
+		 
+		var cell = row.firstChild.firstChild;
+		cell.setAttribute('value', indexOfId != -1);
+	},
+	
+	
+	initSyncedLibraries: Zotero.Promise.coroutine(function* () {
+		var apiKey = Zotero.Sync.Data.Local.getAPIKey();
+		var client = Zotero.Sync.Runner.getAPIClient({apiKey});
+		var keyInfo = yield Zotero.Sync.Runner.checkAccess(client, {timeout: 5000});
+		var tree = document.getElementById("synced-libraries-tree");
+		var treechildren = document.getElementById('synced-libraries-rows');
+		while (treechildren.hasChildNodes()) {
+			treechildren.removeChild(treechildren.firstChild);
+		}
+		
+		function addRow(libraryName, id, checked=false, editable=true) {
+			var treeitem = document.createElement('treeitem');
+			var treerow = document.createElement('treerow');
+			var checkboxCell = document.createElement('treecell');
+			var nameCell = document.createElement('treecell');
+			
+			nameCell.setAttribute('label', libraryName);
+			nameCell.setAttribute('value', id);
+			nameCell.setAttribute('editable', false);
+			checkboxCell.setAttribute('value', checked);
+			checkboxCell.setAttribute('editable', editable);
+			
+			treerow.appendChild(checkboxCell);
+			treerow.appendChild(nameCell);
+			treeitem.appendChild(treerow);
+			treechildren.appendChild(treeitem);
+		}
+		
+		// Add an animated row for while we're loading a group list
+		var loadingLabel = Zotero.getString("zotero.preferences.sync.syncedLibraries.loadingLibraries");
+		addRow(loadingLabel, "loading", false, false);
+		var cell = treechildren.firstChild.getElementsByAttribute("value", "loading")[0];
+		function animateLoadingLabel() {
+			if (cell.getAttribute('label').length < loadingLabel.length + 3) {
+				cell.setAttribute('label', cell.getAttribute('label') + '.');
+			} else {
+				cell.setAttribute('label', loadingLabel);
+			}
+			loadTimeout = setTimeout(animateLoadingLabel, 1000.0/3.0);
+		}
+		var loadTimeout = setTimeout(animateLoadingLabel, 1000.0/3.0);
+
+		// Add default rows
+		addRow(Zotero.getString("pane.collections.library"), null, true, false);
+		addRow(Zotero.getString("pane.collections.publications"), null, true, false);
+		addRow(Zotero.getString("pane.collections.feeds"), null, true, false);
+
+		var syncedLibraries = JSON.parse(Zotero.Prefs.get('sync.syncedLibraries') || '[]');
+		
+		// Load up remote groups
+		var groupIDs = yield client.getGroupVersions(keyInfo.userID);
+		for (let groupID in groupIDs) {
+			let group = yield client.getGroupInfo(groupID);
+			addRow(group.data.name, group.id, syncedLibraries.indexOf(group.id) != -1);
+		}
+		
+		// Remove the animated row
+		clearTimeout(loadTimeout);
+		treechildren.removeChild(treechildren.firstChild);
```

Comment:
This temporary line doesn't really work — it doesn't make sense (because there are already libraries displayed) and makes everything jump around. If we were going to do a temp line in the list this way, it should say "Loading groups…" and be in the place where the groups will go.

But since we can remove the individual group calls, I'd say we should either not show a temporary line at all (since most of the time the loading will be very quick) or not show any library lines at first and show "Loading libraries…". I'd be inclined to do the former.